test/hotspot/jtreg/serviceability/jvmti/VMEvent/libVMEventTest.c

```diff
@@ -47,12 +47,36 @@ extern "C" {
 
 #endif
 
+// JVMTI_ERROR_WRONG_PHASE guard
+static jrawMonitorID event_mon = NULL;
+static int is_vm_dead = 0;
+
+void lock(jvmtiEnv *jvmti, JNIEnv *jni) {
+  jvmtiError err = (*jvmti)->RawMonitorEnter(jvmti, event_mon);
+  if (err != JVMTI_ERROR_NONE) {
+    JNI_ENV_PTR(jni)->FatalError(JNI_ENV_ARGS2(jni, "RawMonitorEnter failed"));
+  }
+}
+
+void unlock(jvmtiEnv *jvmti, JNIEnv *jni) {
+  jvmtiError err = (*jvmti)->RawMonitorExit(jvmti, event_mon);
+  if (err != JVMTI_ERROR_NONE) {
+    JNI_ENV_PTR(jni)->FatalError(JNI_ENV_ARGS2(jni, "RawMonitorExit failed"));
+  }
+}
+
```

Comment:
Nit: It'd be better to use the `RawMonitorLocker` defined in the `jvmti_common.hpp`.
The usages can be found in the `test/hotspot/jtreg/serviceability/jvmti/vthread` tests.

Reply:
Sorry, my suggestion is no applicable. The problem is that this is not a c++ file.